mil_common/drivers/mil_passive_sonar/scripts/triggering

```diff
@@ -0,0 +1,228 @@
+#!/usr/bin/env python
+import rospy
+import numpy as np
+from rospy.numpy_msg import numpy_msg
+from std_msgs.msg import Header
+from std_srvs.srv import SetBool, SetBoolResponse
+from std_srvs.srv import Trigger, TriggerResponse
+from mil_passive_sonar.msg import HydrophoneSamples, Ping2
+from mil_passive_sonar.srv import SetTargetFrequency, SetTargetFrequencyResponse
+from mil_passive_sonar import util
+from mil_ros_tools import Plotter, interweave
+
+from scipy.signal import fftconvolve
+from scipy.ndimage.filters import maximum_filter1d
+
+
+class HydrophoneTrigger:
+
+    def __init__(self):
+        '''ROS Node meant to trigger only when a ping happens in our target frequency range
+
+        Subscribes to raw hydrophone samples on `samples`
+
+        publishes samples(of gradient) from right around the triggering on `pings` when a ping is detected
+            and found to be in the target frequency range
+
+        publish a plot of the frequency response of the filter on `/filter_debug`  by service calling `/filter_debug_trigger`
+
+        '''
+
+        # Ros stuff
+        rospy.init_node('hydrophone_trigger')
+        self.sub = rospy.Subscriber('samples', numpy_msg(HydrophoneSamples), self.hydrophones_cb)
+        self.pub = rospy.Publisher('pings', numpy_msg(Ping2), queue_size=1)
+        rospy.Service('~filter_debug_trigger', Trigger, self.filter_response)
+        rospy.Service('~enable', SetBool, self.enable)
+
+        self.enabled = True
+
+        # Attriburtes about our general frequency range (all pinger live here)
+        #  Frequency range garunteed to be relatively quiet except for the pingers (in Hz)
+        self.general_lower = 15000 # lowest frequency pinger - 10 kHz
+        self.general_upper = 50000 # highest frequency pinger + 10 kHz
+
+        # Attributes about our target frequency range
+        #  target Frquency in Hz
+        self.target = rospy.get_param('~target_frequency')
```

Comment:
So we dropped this being configurable via a service? We will want missions to be able to on the fly change this without restarting this node. I could imagine it being an alias like `set_pinger_freq 25000`

Reply:
run time configuration now supported, see docs